Extension/src/LanguageServer/configurations.ts

```diff
@@ -1896,9 +1896,7 @@ export class CppProperties {
         }
 
         // Resolve and split any environment variables
-        paths = this.resolveAndSplit(paths, undefined, this.ExtendedEnvironment);
-        compilerPath = util.resolveVariables(compilerPath, this.ExtendedEnvironment).trim();
-        compilerPath = this.resolvePath(compilerPath);
+        compilerPath = this.resolvePath(compilerPath).trim();
```

Comment:
technically the `trim` goes on the input, not the output to keep the same behavior as before.  I don't know if it actually matters, but my rule of thumb is to change as little as possible to avoid unforeseen regressions.

```ts
compilerPath = this.resolvePath(compilerPath.trim());
```

Reply:
I take that back.  This changes the behavior too, just differently.  Feel free to ignore.